templates/tab/js/m365/src/components/sample/Welcome.jsx

```diff
@@ -0,0 +1,77 @@
+import React, { useState } from "react";
+import { Image, Menu } from "@fluentui/react-northstar";
+import "./Welcome.css";
+import { EditCode } from "./EditCode";
+import { AzureFunctions } from "./AzureFunctions";
+import { Graph } from "./Graph";
+import { CurrentUser } from "./CurrentUser";
+import { useTeamsFx } from "./lib/useTeamsFx";
+import { TeamsFx } from "@microsoft/teamsfx";
+import { useData } from "./lib/useData";
+import { Deploy } from "./Deploy";
+import { Publish } from "./Publish";
+
+export function Welcome(props) {
+  const { showFunction, environment } = {
+    showFunction: true,
+    environment: window.location.hostname === "localhost" ? "local" : "azure",
+    ...props,
+  };
+  const friendlyEnvironmentName =
+    {
+      local: "local environment",
+      azure: "Azure environment",
+    }[environment] || "local environment";
+
+  const steps = ["local", "azure", "publish"];
+  const friendlyStepsName = {
+    local: "1. Build your app locally",
+    azure: "2. Provision and Deploy to the Cloud",
+    publish: "3. Publish to Teams",
+  };
+  const [selectedMenuItem, setSelectedMenuItem] = useState("local");
+  const items = steps.map((step) => {
+    return {
+      key: step,
+      content: friendlyStepsName[step] || "",
+      onClick: () => setSelectedMenuItem(step),
+    };
+  });
+
+  const { isInTeams } = useTeamsFx();
+  const userProfile = useData(async () => {
+    const teamsfx = new TeamsFx();
+    return isInTeams ? await teamsfx.getUserInfo() : undefined;
+  })?.data;
+  const userName = userProfile ? userProfile.displayName : "";
+  return (
+    <div className="welcome page">
+      <div className="narrow page-padding">
+        <Image src="hello.png" />
+        <h1 className="center">Congratulations{userName ? ", " + userName : ""}!</h1>
+        <p className="center">Your app is running in your {friendlyEnvironmentName}</p>
```

Comment:
I suggest to add the hub name info here, to demo the functionality of MetaOS SDK.

Reply:
Agree, but let me add it in another PR for easy code review.